packages/clerk-js/src/ui/elements/Pagination.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useState } from 'react';
+
+import { Button, Flex, localizationKeys, Text } from '../customizables';
+import { PropsOfComponent } from '../styledSystem';
+import { range } from '../utils';
+
+type UsePaginationProps = {
+  defaultPage?: number;
+};
+
+export const usePagination = (props?: UsePaginationProps) => {
+  const { defaultPage = 1 } = props || {};
+  const [page, setPage] = useState(defaultPage);
+
+  return { page, changePage: setPage };
+};
+
+const PageButton = (props: PropsOfComponent<typeof Button> & { isActive?: boolean }) => {
+  const { sx, isActive, ...rest } = props;
+
+  return (
+    <Button
+      size='xs'
+      variant='ghost'
+      colorScheme='neutral'
+      sx={t => [
+        {
+          color: isActive ? t.colors.$black : t.colors.$blackAlpha700,
+        },
+        sx,
+      ]}
+      {...rest}
+    />
+  );
+};
+
+type RowInfo = {
+  startingRow: number;
+  endingRow: number;
+  allRowsCount: number;
+};
+
+type RowInfoProps = {
+  rowInfo: RowInfo;
+};
+
+const RowInformation = (props: RowInfoProps) => {
+  const {
+    rowInfo: { startingRow, endingRow, allRowsCount },
+  } = props;
+
+  return (
+    <Text>
```

Comment:
❓ Is the outer `<Text/>` required?

Reply:
I think I put it there so the component renders a block.